logbook-core/src/main/java/org/zalando/logbook/BodyFilters.java

```diff
@@ -34,6 +35,14 @@ public static BodyFilter accessToken() {
         return replaceJsonStringProperty(properties, "XXX");
     }
 
+    @API(status = MAINTAINED)
+    public static BodyFilter oauthRequest() {
```

Comment:
according to [client-credentials](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/) and [password](https://www.oauth.com/oauth2-servers/access-tokens/password-grant/) flows those parameters looks quite standardised. Also I think it makes sense to add filtering out `refresh_token` form `accessToken()` cause it in the end have much higher risks that `access_token`

Reply:
according to [w3..org](https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1): `x-ww-form-urlencoded` should follow [RFC-1738](https://www.ietf.org/rfc/rfc1738.txt), that describes  URL formats.

Reply:
> Also I think it makes sense to add filtering out `refresh_token` form `accessToken()` cause it in the end have much higher risks that `access_token`

Agreed!

Reply:
> [RFC-1738](https://www.ietf.org/rfc/rfc1738.txt), that describes URL formats.

That one doesn't describe query parameters, does it?

Reply:
as it stays in rfc `RFC 1738            Uniform Resource Locators (URL)        December 1994`, so it is describes query parameters, but I am not able to say, is it describe a latest state.

Reply:
I don't find anything query parameter related in there.

Reply:
I would agree that rfc is quite unreadable. Therefore there is a part describes encoding part as well as format of HTTP URL

```
Unsafe:

   Characters can be unsafe for a number of reasons.  The space
   character is unsafe because significant spaces may disappear and
   insignificant spaces may be introduced when URLs are transcribed or
   typeset or subjected to the treatment of word-processing programs.
   The characters "<" and ">" are unsafe because they are used as the
   delimiters around URLs in free text; the quote mark (""") is used to
   delimit URLs in some systems.  The character "#" is unsafe and should
   always be encoded because it is used in World Wide Web and in other
   systems to delimit a URL from a fragment/anchor identifier that might
   follow it.  The character "%" is unsafe because it is used for
   encodings of other characters.  Other characters are unsafe because
   gateways and other transport agents are known to sometimes modify
   such characters. These characters are "{", "}", "|", "\", "^", "~",
   "[", "]", and "`".

   All unsafe characters must always be encoded within a URL. For
   example, the character "#" must be encoded within URLs even in
   systems that do not normally deal with fragment or anchor
   identifiers, so that if the URL is copied into another system that
   does use them, it will not be necessary to change the URL encoding.
```

```
; HTTP

httpurl        = "http://" hostport [ "/" hpath [ "?" search ]]
hpath          = hsegment *[ "/" hsegment ]
hsegment       = *[ uchar | ";" | ":" | "@" | "&" | "=" ]
search         = *[ uchar | ";" | ":" | "@" | "&" | "=" ]
```

Reply:
For example [RFC2396](https://www.ietf.org/rfc/rfc2396.txt) mentioned in `java.net.URL` provides almost same idea 
```
2. URI Characters and Escape Sequences

   URI consist of a restricted set of characters, primarily chosen to
   aid transcribability and usability both in computer systems and in
   non-computer communications. Characters used conventionally as
   delimiters around URI were excluded.  The restricted set of
   characters consists of digits, letters, and a few graphic symbols
   were chosen from those common to most of the character encodings and
   input facilities available to Internet users.

      uric          = reserved | unreserved | escaped

   Within a URI, characters are either used as delimiters, or to
   represent strings of data (octets) within the delimited portions.
   Octets are either represented directly by a character (using the US-
   ASCII character for that octet [ASCII]) or by an escape encoding.
   This representation is elaborated below.
```

```
3.4. Query Component

   The query component is a string of information to be interpreted by
   the resource.

      query         = *uric

   Within a query component, the characters ";", "/", "?", ":", "@",
   "&", "=", "+", ",", and "$" are reserved.
```

Reply:
So, how I personally understand it. form url encoded bodies should follow same RFCs, those describe URL format. Both (URI and URL) RFCs describes same for query|search part.

Reply:
Guys, what about `refresh_token`? Did you consciously decide **not** to filter it? I just stumbled upon the fact that the refresh token appears in our logs....